### ISSUE TITLE
Make it an operator for more fine-grained permissions

### ISSUE THREAD

**Issue by k** [ksync/ksync#394]
If ksync adds an operator that ran on the cluster and watch for a CRD with `kind: Ksync`, then we could use the similar permissions that are given to devs on the cluster. 

This is not a rigorous design, but what I would imagine is when the operator detects a `Ksync` resource, it deploys a pod that ksync uses to proxy to send information to the operator that uses the ksync daemonset to fulfill the syncing.

The `Ksync` resource would have a spec that adds the target path and selector as well as desired client hostpath and some uniquely identifying information for the client so it can't be attempting to be synced by two clients at once.

**Comment by grampelberg** [comment]
This is a cool idea! You wouldn't even need an operator to do it. Just add an APIService and RBAC on top of that to manage access to `radar` and you'd be set.

FWIW, the alternative approach here is to lean on EphemeralContainers and `shareProcessNamespace`.

**Comment by k** [comment]
Yep the `APIService` sounds like a cleaner solution, operator is a bit heavy for this purpose.

Ephemeral Containers sounds even cleaner though! This means we wouldn't need the DaemonSet anymore. Radar wouldn't be necessary anymore at all then right? ksync becomes client only.

**Comment by grampelberg** [comment]
Yup! Ksync would be 100% client only. I've been wanting to do the restructuring, just not had the time.

**Comment by k** [comment]
Happy to help @grampelberg ! How far are you along with it? Would starting a design doc be helpful?

**Comment by grampelberg** [comment]
I've done nothing outside of "I bet this would all work together!". A design doc would be awesome!! @timfallmk can help out here too =)
